vowpalwabbit/parse_example_json.h

```diff
@@ -1080,8 +1081,29 @@ class DecisionServiceState : public BaseState<audit>
     return this;
   }
 
-  BaseState<audit>* EndObject(Context<audit>& /* ctx */, rapidjson::SizeType /* memberCount */) override
+  BaseState<audit>* EndObject(Context<audit>& ctx, rapidjson::SizeType /* memberCount */) override
   {
+    // If we are in CCB mode and there have been no decisions. Check label cost, prob and action were passed. In that
+    // case this is CB, so generate a single decision with this info.
+    auto jsonp = static_cast<json_parser<audit>*>(ctx.all->p->jsonp.get());
+    if (jsonp->mode == json_parser_mode::ccb)
+    {
+      auto num_decisions = std::count_if(ctx.examples->begin(), ctx.examples->end(),
+          [](example* ex) { return ex->l.conditional_contextual_bandit.type == CCB::example_type::decision; });
+      if (num_decisions == 0 && ctx.label_object_state.found_cb)
```

Comment:
Maybe we should handle inconsistency (num_decisions != 0 && found_cb) somehow (throw / report error)?

Reply:
found_cb is actually set in the ccb path, it is set for and _Label_... field that is encountered. I agree it's not great but is the case because of the natural overlap